projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -6,26 +6,110 @@
 
 #include <boost/asio.hpp>
 #include <string>
+#include <iterator>
 #include <iostream>
-#include <cstdint> //uint8_t, uint32_t
 
 using STG::RADIUS;
+using STG::RAD_SETTINGS;
 
 extern "C" STG::Plugin* GetPlugin()
 {
     static RADIUS plugin;
     return &plugin;
 }
 
-std::string RADIUS::GetVersion() const
+RAD_SETTINGS::RAD_SETTINGS()
+    : m_port(0)
+{}
+
+int RAD_SETTINGS::ParseSettings(const ModuleSettings & s)
 {
-    return "Radius v.1.0";
+    ParamValue pv;
+    int p;
+
+    pv.param = "Port";
+    auto pvi = std::find(s.moduleParams.begin(), s.moduleParams.end(), pv);
+    if (pvi == s.moduleParams.end() || pvi->value.empty())
+    {
+        m_errorStr = "Parameter \'Port\' not found.";
+        printfd(__FILE__, "Parameter 'Port' not found\n");
+        return -1;
+    }
+    if (ParseIntInRange(pvi->value[0], 2, 65535, &p) != 0)
+    {
+        m_errorStr = "Cannot parse parameter \'Port\': " + m_errorStr;
+        printfd(__FILE__, "Cannot parse parameter 'Port'\n");
+        return -1;
+    }
+    m_port = static_cast<uint16_t>(p);
+
+    pv.param = "Secret";
+    pvi = std::find(s.moduleParams.begin(), s.moduleParams.end(), pv);
+    if (pvi == s.moduleParams.end() || pvi->value.empty())
+    {
+        m_errorStr = "Parameter \'Secret\' not found.";
+        printfd(__FILE__, "Parameter 'Secret' not found\n");
+        m_secret = "";
+    }
+    else
+    {
+        m_secret = pvi->value[0];
+    }
+
+    pv.param = "Dictionaries";
+    pvi = std::find(s.moduleParams.begin(), s.moduleParams.end(), pv);
+    if (pvi == s.moduleParams.end() || pvi->value.empty())
+    {
+        m_errorStr = "Parameter \'Dictionaries\' not found.";
```

Comment:
This parameter is also optional, so it is not an error when the parameter is missing.